error TS2614: Module '"@fastify/fast-json-stringify-compiler"' has no exported member 'SerializerFactory'

### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure the bug has not already been reported


### Fastify version

4.14

### Plugin version

_No response_

### Node.js version

v16.19.1

### Operating system

Linux

### Operating system version (i.e. 20.04, 11.3, 10)

22.04.2

### Description

Hi,

running Typescripts tsc  (v4.9.5) from GitHub CI results in:
```
Error: node_modules/fastify/fastify.d.ts(8,33): error TS2614: Module '"@fastify/fast-json-stringify-compiler"' has no exported member 'SerializerFactory'. Did you mean to use 'import SerializerFactory from "@fastify/fast-json-stringify-compiler"' instead?
Error: node_modules/fastify/types/schema.d.ts(2,10): error TS2[6](https://github.com/seriousme/fastify-openapi-glue/actions/runs/4350791294/jobs/7748670225#step:11:7)14: Module '"@fastify/fast-json-stringify-compiler"' has no exported member 'SerializerFactory'. Did you mean to use 'import SerializerFactory from "@fastify/fast-json-stringify-compiler"' instead?
Error: Process completed with exit code 2.
```

I checked  @fastify/fast-json-stringify-compiler and it looks to me like SerializerFactory tries to refer to:
`type FastJsonStringifyFactory = () => SerializerSelector.SerializerFactory`
https://github.com/fastify/fast-json-stringify-compiler/blob/a068619bafa4e2e8c449e6056a20f28a7711b4cd/types/index.d.ts#L3

Kind regards,
Hans


### Steps to Reproduce

See https://github.com/seriousme/fastify-openapi-glue/pull/461
4.13.0 has no issues,  4.14.0 and 14.14.1 break my CI 

### Expected Behavior

No error

--- a/package.json
+++ b/package.json
@@ -180,7 +180,7 @@
   "dependencies": {
     "@fastify/ajv-compiler": "^3.5.0",
     "@fastify/error": "^3.0.0",
-    "@fastify/fast-json-stringify-compiler": "^4.1.0",
+    "@fastify/fast-json-stringify-compiler": "^4.2.0",
     "abstract-logging": "^2.0.1",
     "avvio": "^8.2.0",
     "fast-content-type-parse": "^1.0.0",

